FirstAnalysis/HFmcvalidation.py

```diff
@@ -1,24 +1,150 @@
-#!/usr/bin/env python
-from ROOT import TCanvas, TFile
-
-
-def p_diff_reco_MC(var):
-    fileo2 = TFile("../codeHF/AnalysisResults_O2.root")
-    cpt = TCanvas("cpt", "pT distribution")
-    cpt.SetCanvasSize(1500, 700)
-    cpt.Divide(2, 1)
-    csecv = TCanvas("csecv", "sec vtx distribution")
-    csecv.SetCanvasSize(1500, 700)
-    csecv.Divide(2, 1)
-    hp = fileo2.Get("hf-mc-validation-rec/histP%s" % var)
-    hsecv = fileo2.Get("hf-mc-validation-rec/histSecV%s" % var)
+from ROOT import TCanvas, TFile, TGaxis, TLegend, gStyle
+
+
+def count_HFquarks_perColl(var):
+    fileo2 = TFile("../codeHF/CCbar_pp_AnalysisResults_O2_500files.root")
+    cquark = TCanvas("cquark", "%s per collision" % var)
+    cquark.SetCanvasSize(900, 700)
+    cquark.Divide(2, 1)
+    hq = fileo2.Get("hf-mc-validation-gen/hCountAverage%s" % var)
+    hq.GetXaxis().SetRangeUser(0.0, 8.0)
+    hq.SetTitle("Number of %s quarks per collision" % var)
+    hq.Draw("")
+    cquark.SaveAs("QuarkCountsPerCollision%s.pdf" % var)
+    cquark.SaveAs("QuarkCountsPerCollision%s.root" % var)
+    cquark.SaveAs("QuarkCountsPerCollision%s.png" % var)
```

Comment:
This is wasting space.
It should be in a function, that takes the object and the name and saves that object for all needed formats in a loop.